docs/proposals/01-storage-encryption.md

```diff
@@ -0,0 +1,147 @@
+- Encrypt block device (RBD image) that CEPH exposes. 
+
+
+### Non-Goals
+- OSD level encryption
+- Object storage encryption
+
+## Proposal
+
+### Way 1:
+### Details
+This way emphasizes more on one level down to abstraction.
+
+- To perform RBD encryption directly at ceph level is by using the LUKS encryption technology, which is built into the Linux kernel and can be used to encrypt block       devices such as RBD. 
+
+- The new RBD image can be created with RBD command line tool or directly from ceph dashboard. LUKS container on the RBD image can be created using `cryptsetup` tool       which can be further mapped to a block device on the client system.
+
+- To use the LUKS format, start by formatting the image:
+
+   ` $ rbd encryption format {pool-name}/{image-name} {luks1|luks2} {passphrase-file} [–cipher-alg {aes-128 | aes-256}] `
+ 
+- A file system is created on the mapped block device and is mounted as per the requirement. Mounted file system is the point of contact to write or read the data from     encrypted RBD image.
+
+- To mount a LUKS-encrypted image run:
+
+   ` $ rbd -p {pool-name} device map -t nbd -o encryption-format={luks1|luks2},encryption-passphrase-file={passphrase-file} `
+
+- When writing data to the encrypted RBD image, it is automatically encrypted by LUKS before being written to the RBD image. When reading data from the RBD image, it       is decrypted by LUKS on the client system before being made available to the application.
+
+### Way 2:
+### Details
+ As of now two types of encryption is supported by Ceph: 
+- OSD Level and Block device Level
+(Currently we are moving ahead in this proposal with Block device level.)
+
+- Data encryption key(DEK) will be provided by user in volume object. Cephlet will internally create a Key encryption key(KEK), encrypt the data encryption key and add     into the metadata. All the changes will take place in cephCSI.
+
+- Currently when `volume` is created corresponding `PVC` is also created with reference to `storageclass`.
+    However, With `Encryption` enabled `Volume` there will be a new storageclass named `encrypted-ceph` will be created which will be referenced while 
+    creating encrypted PVCs with cephlet generated passphrase. 
+ 
+- Encrypted storage class will be created with additional parameters as below.
+
+```
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: rook-ceph-csi-kms-config
+  namespace: rook-ceph
+data:
+  config.json: |-
+    {
+      "user-secret-metadata": {
+        "encryptionKMSType": "metadata",
+        "secretName": "storage-encryption-secret"
+      }
+    }
+```
+
+Additionally Update the rook-ceph-operator-config configmap and patch the following configurations
+```
+kubectl patch cm rook-ceph-operator-config -nrook-ceph -p $'data:\n "CSI_ENABLE_ENCRYPTION": "true"'
+```
+
+
+```
+apiVersion: v1
```

Comment:
Maybe examples could be expanded to also include how the DEK and KEK are stored.

Reply:
We have expanded the examples to include volume object which has secret reference to DEK. 
KEK is generated by cephlet and example we have added in proposal. 

FYI, currently the proposal is in revision state and we might need to restructure it according to final decisions.